src/metabase/public_settings.clj

```diff
@@ -627,10 +630,23 @@
   :type       :boolean
   :default    false)
 
+(defn- not-handling-api-request?
+  []
+  (nil? @api/*current-user*))
+
+(defn set-uploads-database-id!
+  "Sets the :uploads-database-id setting, with an appropriate permission check."
+  [new-id]
+  (if (or (not-handling-api-request?)
```

Comment:
Is this setting ever set in a context where a user isn't bound? If not, I think it'd be better to omit this check

Reply:
I wrote this for tests, but it'd be convenient for REPL usage too.